lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -210,17 +213,31 @@ def test_no_ccx_created(self):
     def test_create_ccx(self):
         """
         Create CCX. Follow redirect to coach dashboard, confirm we see
-        the coach dashboard for the new CCX.
+        the coach dashboard for the new CCX. Also, confirm that the coach
+        has staff access to it.
         """
+
         self.make_coach()
         url = reverse(
             'create_ccx',
-            kwargs={'course_id': self.course.id.to_deprecated_string()})
+            kwargs={'course_id': unicode(self.course.id)})
+
         response = self.client.post(url, {'name': 'New CCX'})
         self.assertEqual(response.status_code, 302)
         url = response.get('location')  # pylint: disable=no-member
         response = self.client.get(url)
         self.assertEqual(response.status_code, 200)
+
+        # To locate the ccx_key, first get the path from the URL
+        path = urlparse.urlparse(url).path
+        
+        # Then get the second to last segment of the path
+        head, tail = os.path.split(path)
+        head, ccx_key = os.path.split(head)
```

Comment:
I think I still need to parse the path out of the url using `urlparse`, otherwise `resolve(url)` returns a 404 (since the url contains `http://testserver/`). But with that done, `resolve(path)` works nicely on the path. Thanks for your help with this, @ormsbee and @sarina. I just pushed a change for your review.